pkg/kubelet/kubelet.go

```diff
@@ -1943,6 +1944,8 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 		if kl.podWorkers.CouldHaveRunningContainers(pod.UID) && !kubetypes.IsStaticPod(pod) {
 			pod = kl.handlePodResourcesResize(pod)
 		}
+	} else {
+		pod = kl.podResizeInfeasible(pod)
```

Comment:
@vinaykul I wanted to ask you if it is really what we want?
I looked into the code once again and it seems fishy for me.

We set `resize = infeasable` if the feature gate is disabled but we don't check if any resize is actually going on (sth like `!cmp.Equal(container.Resources.Requests, containerStatus.AllocatedResources)`). 

Maybe I'm missing sth, but I think we should add a check if the `Resources.Requests` is not equal to `AllocatedResources` indicating there is request for a change). 

Is it right?

Reply:
I see your point, there are other cases to consider. For this issue, I want to focus on `InPlacePodVerticalScaling` feature gate is disabled on kubelet when it sees `PodStatus.Resize=Proposed` (i.e it was enabled on apiserver and a resize was requested).
The other key scenario that you are worried about is when user previously enabled InPlacePodVerticalScaling` feature gate and resize is in-flight. Perhaps we can create a new issue for that if tests prove there is a problem? (There is already a bunch of ugly special code for container hash handling during the alpha/beta transition period. I just want to know how big of a problem it is and how much special code effort we want to invest into it that will get dropped once we go GA)